src/SonarQube.TeamBuild.PostProcessor/MSBuildPostProcessor.cs

```
@@ -83,10 +84,15 @@ public bool Execute(string[] args, AnalysisConfig config, ITeamBuildSettings set
                 return false;
             }
 
+            if (!CheckCredentialsInCommandLineArgs(config, provider))
```

Not sure whether this check should be done a bit earlier or later in this method.

---

I think it should be; if we're going to fail, we should fail fast before doing any other work.

---

IMO we have 3 choices:
1. Where it currently is

---

I think the check is fine here (although I'd probably have put it before the _CheckEnvironmentConsistency_ simply because it's a more common scenario. It makes no difference to the outcome, but I tend to find code easier to follow if the major/common scenarios are handled before the corner cases).